### ISSUE TITLE
puzzle piece missing after exiting customization

### ISSUE THREAD

**Issue by jmain33** [Quicksaver/Puzzle-Toolbars#95]
#### The puzzle piece in question is either gone or hidden behind the puzzle panel.

1. Conditions for Reproduction: The ***corner puzzle bar*** must be showing prior to entering customization…
2. Settings Used: show piece, positioned on right, not set to auto-hide, not extended

Recovering the piece can be achieved by toggling a setting on the *corner piece's* option page or by using the shortcut [CTRL][F2] to hide the bar and clicking the piece which, as it seems, was actually still present.
***
BTW: You should adjust your wording in the options pages. The keyboard shortcuts aren't for *[enabling/disabling]* but instead to **[show/hide]**.

**Comment by Quicksaver** [comment]
> The puzzle piece in question is either gone or hidden behind the puzzle panel.

I can't reproduce this at all, the puzzle piece works (and shows) fine for me after I exit customization. Can you give me some exact steps to reproduce it? What operating system and Firefox version are you on?

> BTW: You should adjust your wording in the options pages. The keyboard shortcuts aren't for [enabling/disabling] but instead to [show/hide].

True.

**Comment by Quicksaver** [comment]
> BTW: You should adjust your wording in the options pages. The keyboard shortcuts aren't for [enabling/disabling] but instead to [show/hide].

Well, actually it's for opening and closing, showing and hiding is yet another thing (i.e. auto-hiding). But I understood what you meant. ;)

**Comment by Quicksaver** [comment]
Does the customization problem still happen with [beta version 2.1.15b1](https://github.com/Quicksaver/Puzzle-Toolbars/releases/tag/v2.1.15b1)?

**Comment by jmain33** [comment]
`Win7 Pro 64-bit`, `Firefox 47.0.1`, `Puzzle Bars 2.1.15b1`  
Still happening sadly :cry: 
> @Quicksaver said
I can't reproduce this at all, the puzzle piece works (and shows) fine for me after I exit customization. Can you give me some exact steps to reproduce it? What operating system and Firefox version are you on?

The conditions and settings I gave in the first post ~~are really all I can think to say~~; plus the only other active puzzle bars are in the **location bar** and **top-most**, neither of which should be causing any interference with the **corner bar**.
*Here's what the corner bar looks like in the customization before exiting*
![2016-10-21_155646](https://cloud.githubusercontent.com/assets/364460/19610465/a974fdaa-97aa-11e6-8607-14ec80ecaf6b.png)

When I exit the customization view the <ins>corner puzzle piece</ins> (all by itself) is displayed for a split second, after which its corresponding bar appears and overlaps it, making the <ins>puzzle piece</ins> inaccessible.
*Screenshot Below • the piece would normally be seen on the right end*
![2016-10-21_155924.png](https://cloud.githubusercontent.com/assets/364460/19610261/84440694-97a9-11e6-9873-02a9f0295ebb.png "The Corner Puzzle Bar After Exiting Customization")
> @jmain33 said
Recovering the piece can be achieved by toggling a setting on the corner piece's option page or by using the shortcut [CTRL][F2] to hide the bar and clicking the piece which, as it seems, was actually still present.

I tried removing the two rather unique toolbar buttons <tt>**search field**</tt> & <tt>**zoom page**</tt> to see if they were at fault; they weren't the cause.

**Comment by Quicksaver** [comment]
Does that happen if you disable all your other add-ons except for Puzzle Bars?

**Comment by jmain33** [comment]
#### Well that was torture😱!! I spent my whole weekend trying to regain control of Firefox.

Disabled every add-on except PB and the problem disappeared… Re-enabled every restart-less add-on and things were still going well… Enabled the least likely culprits, restarted and PB still showed no symptom of this… Now down to enabling the last _≈5_ or so add-ons (out of 55) and Firefox **won't enable them**; it's crashing <ins>on every single exit</ins>.

Gladly😌, it wasn't all for naught😤, I pinpointed [Status-4-Evar](https://github.com/SparkyBluefang/S4E) as the abusive mate. It's also responsible for forcing the dropper to be placed behind the location puzzle bar.

These are very minor issues though; S4E **can**, for the most part, safely be used with PB. Still, <ins>_I've grudgingly decided not to ⇒ and disabled S4E._</ins>

• <samp>the :zap:crashing persists more-so than ever; I can no longer cleanly exit Firefox.</samp> •
• <ins><samp>Disabling 50+ add-ons at once just isn't great for the clunky old fox.</samp></ins> •
***
### [BOT] I estimate there is nothing to do on your end (it is a S4E issue).

**Comment by Quicksaver** [comment]
I have to admit I don't see the problem with Status-4-Evar either. :( Maybe it was some specific setting in it, but I tried flipping through most of them and I can't get the puzzle icon to disappear after exiting customize mode.

But of course if you choose to disable it then I guess we're done here for now. :) Please do let me know if you ever try to give it another go and if you can see if there's a specific S4E preference combination needed to reproduce it.

> It's also responsible for forcing the dropper to be placed behind the location puzzle bar.

Also, what dropper?

**Comment by jmain33** [comment]
@Quicksaver - The little pale arrow in the location bar that drops down your most recently/frequently visited addresses. I'm just discovering, since I've been using S4E much longer than PB, for the first time that the arrow <ins>should be to the left of my puzzle bar</ins>.

Did you notice this change when testing out S4E?

> …Maybe it was some specific setting in it, but I tried flipping through most of them and…

It has an option to show the current site's loading progress in the location bar. I also used it to show the activity status (such as the currently hovered link's address) on the right side of the location bar, instead of its usual placement in the lower corners.
![S4E in location bar](https://addons.cdn.mozilla.net/user-media/previews/full/55/55258.png?modified=1313340435 "Links and status text can be shown in the location bar. This shows the default configuration with the text aligned right. But it can be set to replace the current URL or be positioned at some fixed spot on the location bar.")
:arrow_heading_down: _screenshots are from AMO; hover for the developer's descriptions_ :arrow_heading_up: 
![S4E in location bar](https://addons.cdn.mozilla.net/user-media/previews/full/55/55259.png?modified=1313340435 "Page load progress can be shown in the location bar (and on a toolbar with the Progress Meter item). Placement, color, and background image can be easily controlled. There is also an advanced style editor that gives you complete access to the CSS background rule.")  
Regarding the corner puzzle bar; the issue only happens when it is placed on the right side, also I do not have any bars showing beneath it (no lower puzzle bar, status bar or add-on bar).

**Comment by Quicksaver** [comment]
> The little pale arrow in the location bar that drops down your most recently/frequently visited addresses. I'm just discovering, since I've been using S4E much longer than PB, for the first time that the arrow should be to the left of my puzzle bar.

Yes it'll do that, it's necessary because of the way S4E implements its features in the location bar. In the end it doesn't really matter since it still functions perfectly and it's still in the location bar, so I don't think there's anything that needs to be done about it.

> Regarding the corner puzzle bar; the issue only happens when it is placed on the right side, also I do not have any bars showing beneath it (no lower puzzle bar, status bar or add-on bar).

Ok, I can reproduce it every once in a while, not always though... It's weird that it does that.

**Comment by Quicksaver** [comment]
And apparently this is not so easily fixed, seems like a bug in the rendering part of things, which I have no control over.

I'll have to figure out a way to get around it. But since it's not really too critical, I'll probably leave it for a later time.

**Comment by jmain33** [comment]
> **<ins>Quote from @Quicksaver</ins>**
And apparently this is not so easily fixed, seems like a bug in the rendering part of things, which I have no control over. I'll have to figure out a way to get around it. But since it's not really too critical, I'll probably leave it for a later time.

Completely understandable 😃
> **<ins>Quote from @Quicksaver</ins>**
Yes it'll do that, it's necessary because of the way S4E implements its features in the location bar. In the end it doesn't really matter since it still functions perfectly and it's still in the location bar, so I don't think there's anything that needs to be done about it.

> **<ins>Quote from @jmain33</ins>**
These are very minor issues though; S4E can, for the most part, safely be used with PB. Still, I've grudgingly decided not to ⇒ and disabled S4E. I estimate there is nothing to do on your end (it is a S4E issue).

As I implied above, after realizing these two issues were caused by Status-4-Evar; I considered it to not be your _(or Puzzle Bars')_ concern. *I only mentioned the location bar tidbit upon discovering it was supposed to show like that by default (read without S4E).* Please, disregard any obligation to fix them.
